api/v1beta1/gcpmachine_types.go

```diff
@@ -302,6 +312,13 @@ type GCPMachineSpec struct {
 	// +optional
 	Preemptible bool `json:"preemptible,omitempty"`
 
+	// ProvisioningModel defines if instance is spot.
+	// If set to "Standard" while preemptible is true, then the VM will be of type "Preemptible".
+	// If "Spot", VM type is "Spot". When unspecified, defaults to "Standard".
+	// +kubebuilder:validation:Enum=Standard;Spot
+	// +optional
```

Comment:
nit: Technically this is fine since it is alphabetical but if any other instance types come up in the future this could change. 

```suggestion
        // +kubebuilder:default=Standard
	// +optional
```